File: CPP/Program/Status.cpp
#pragma once
#include <iostream>
using namespace std;

class Status {
private:
    int HP;        // Menyimpan nilai HP (Health Point)
    int Stamina;   // Menyimpan nilai stamina
    int Level;     // Menyimpan level karakter
public:
    // Konstruktor dengan nilai default
    Status(int hp = 100, int stamina = 50, int level = 1)
        : HP(hp), Stamina(stamina), Level(level) {}

    // Setter
    void setHP(int hp) { HP = hp; }
    void setStamina(int s) { Stamina = s; }
    void setLevel(int l) { Level = l; }

    // Getter
    int getHP() { return HP; }
    int getStamina() { return Stamina; }
    int getLevel() { return Level; }

    // Menampilkan status ke layar
    void tampilkanStatus() {
        cout << "HP: " << HP
             << ", Stamina: " << Stamina
             << ", Level: " << Level << endl;
    }
};

File: Python/Program/Status.py
class Status:
    def __init__(self, hp=100, stamina=50, level=1):
        # atribut privat pakai double underscore
        self.__hp = hp
        self.__stamina = stamina
        self.__level = level

    # Setter
    def set_hp(self, hp): 
        self.__hp = hp   # ubah nilai HP

    def set_stamina(self, s): 
        self.__stamina = s   # ubah nilai stamina

    def set_level(self, l): 
        self.__level = l   # ubah nilai level

    # Getter
    def get_hp(self): 
        return self.__hp   # ambil nilai HP

    def get_stamina(self): 
        return self.__stamina   # ambil nilai stamina

    def get_level(self): 
        return self.__level   # ambil nilai level

    # tampilkan semua status dalam satu baris
    def tampilkan_status(self):
        print(f"HP: {self.__hp}, Stamina: {self.__stamina}, Level: {self.__level}")
